src/action.js

```diff
@@ -0,0 +1,75 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+
+const format = (obj) => JSON.stringify(obj, undefined, 2);
+
+async function run() {
+  core.info(`
+    *** ACTION RUN - START ***
+    `);
+
+  try {
+    const { payload, eventName } = github.context;
+
+    if (eventName !== "pull_request") {
+      throw new Error(
+        `This action can only run on "pull_request", but "${eventName}" was received. Please check your workflow.`
+      );
+    }
+
+    core.debug(`
+    *** PAYLOAD ***
+    ${format(payload)}
+    `);
+
+    const {
+      number,
+      repository: { owner, name },
+    } = payload;
+
+    const token = core.getInput("github-token");
+    const octokit = github.getOctokit(token);
+    const data = await octokit.graphql(
+      `
+      query getLinkedIssues($owner: String!, $name: String!, $number: Int!) {
+        repository(owner: $owner, name: $name) {
+          pullRequest(number: $number) {
+            closingIssuesReferences {
+              totalCount
+            }
+          }
+        }
+      }
+      `,
+      {
+        owner: owner.login,
+        name,
+        number,
+      }
+    );
+
+    core.debug(`
+    *** GRAPHQL DATA ***
+    ${format(data)}
+    `);
+
+    const linkedIssuesCount =
+      data?.repository?.pullRequest?.closingIssuesReferences?.totalCount;
+
+    core.setOutput("linked_issues_count", linkedIssuesCount);
+
+    if (!linkedIssuesCount) {
+      core.setFailed(
```

Comment:
Just to be sure: do we want to fail if there are zero linked issues?
This will lead to red CI if I'm not wrong, so I would mention the [`continue-on-error`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) setting into the readme

Reply:
Yea good point, perhaps make it configurable? 
I was thinking that the action should post a comment if it finds no linked issues and `setFailed` should be called based on a configurable input (something like `mode: enforce|notify` idk)

Reply:
@salmanm to get this out I think we can go with a simplified implementation:

- let's decide that the default behavior is to fail the build for now
- let's create an issue to make this behavior configurable. In the issue let's also include that we may want to consider implementing this behavior via [check runs](https://docs.github.com/en/rest/reference/checks) (I don't know if it really makes sense but just to research it really)

Reply:
> Just to be sure: do we want to fail if there are zero linked issues? This will lead to red CI if I'm not wrong, so I would mention the [`continue-on-error`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) setting into the readme

I am not sure if that is the effect. I mean this action will need to be used in a standalone workflow because it needs some specific triggers. So failing it will only fail that workflow. If the workflow is not configured as required in the repo settings (which depends on the user to configure), then the overall build will not fail, only this workflow, which will prevent the PR from being merged only if the owner of the repo configured it as a required check (which it isn't by default).

I agree we can document this, but it's basic github knowledge and will not require any "continue-on-error"